src/3d/qgs3d.cpp

```diff
@@ -0,0 +1,32 @@
+/***************************************************************************
+                         qgs3d.cpp
+                         ----------
+    begin                : July 2020
+    copyright            : (C) 2020 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3d.h"
+
+Qgs3D *Qgs3D::instance()
+{
+  static Qgs3D *sInstance( new Qgs3D() );
+  return sInstance;
```

Comment:
Should we use `Q_GLOBAL_STATIC` so it's cleaned up on exit?

Reply:
Possibly? If we apply it to QgsGui too...

Reply:
Worth a try, with a potential revert if we have crashes on exit because of interdependencies.

Reply:
Done